features/steps/validation_handling.py

```diff
@@ -175,8 +175,9 @@ def handle_then(context, fn, **kwargs):
     # ensure the rule is not activated when there are no instances
     # in case there are no instances but the rule is applicable (e.g. SPS001),
     # then the rule is still activated and will return either a pass or an error
+    # an exception is when the feature tags contain '@not-activation'
     is_activated = any(misc.recursive_flatten(instances)) if instances else context.applicable
-    if is_activated:
+    if is_activated and not ('no-activation' in context.tags and context.config.userdata.get('execution_mode') == 'ExecutionMode.PRODUCTION'):
```

Comment:
Why only execution_mode=production?

Reply:
My idea was that this way, in CI/CD and locally, we can still determine `pass` testfiles are actually validated by one of the `Then `statements.

Reply:
Ok, it doesn't really hurt, but I think the idea was that we would keep the behaviour as much the same as possible so this is a bit weird.